shared/lib/formatting/index.js

```diff
@@ -332,8 +335,37 @@ function unescapeAndFormatSpaces( str ) {
 	return decodeEntities( str ).replace( / /g, nbsp );
 }
 
+/**
+ * Traverse an object or array decoding HTML entities on string properties
+ * Accepts a paths argument selecting which properties to update (See lodash.get)
+ *
+ * It does its job in place, so the object passed will be updated.
+ * This helper is used to decode HTML entities in strings properties,
+ * on an object.
+ * We need to decode HTML entities because the REST API returns them
+ * already encoded, and React will encode them again.
+ *
+ * @param	{Object} obj Object to traverse in preparation for React rendering
+ * @param {String/Array} paths Path or paths to decode.
+ */
+function traverseAndDecodeEntities( obj, paths ) {
```

Comment:
Thanks @ebinnion. I understand that `traverseAndDecodeEntity` function mutates `data`. I declared that on its docs for it to be reliable. I also took in account that a straightforward alternative like

```
this.settings = data;
this.settings.display_name = decodeEntities ( data.display_name );
this.settings.user_URL = decodeEntities ( data.user_URL );
this.settings.description = decodeEntities ( data.description );
```
would be also mutating `data`. 

IMHO, this case of a function mutating its argument is particular because this is an anonymous function declared and defined specifically for being passed control by the API request method. Thus it offers no interface and no chance for secondary effects because no others usages of this function are possible. Also, take in account this is the beginning of `this.settings` life cycle.

It seems to me, that unless a full `mapValues()` is done on the whole object, there's no alternative to mutating `data`.

I think that in order for this fix to go along I'll drop that `traverseAndDecodeEntities` and just do this:
```
this.settings = mapvalues( data );
this.settings.display_name = data.display_name && decodeEntities ( data.display_name );
this.settings.user_URL = data.user_URL && decodeEntities ( data.user_URL );
this.settings.description =data.description && decodeEntities ( data.description );
```

If needed, a more general approach or utility function can be added as an independent PR. What do you think?